python_modules/libraries/dagster-looker/dagster_looker/api/dagster_looker_api_translator.py

```diff
@@ -15,6 +20,103 @@ class LookerInstanceData:
     explores_by_id: Dict[str, LookmlModelExplore]
     dashboards_by_id: Dict[str, Dashboard]
 
+    @staticmethod
+    def build(*, sdk: Looker40SDK) -> "LookerInstanceData":
+        """Fetches all explores and dashboards from the Looker instance.
+
+        TODO: Fetch explores in parallel using asyncio
+        TODO: Get all the LookML views upstream of the explores
+        """
+        # Get dashboards
+        dashboards = sdk.all_dashboards()
+        dashboards_by_id = {
+            dashboard.id: sdk.dashboard(dashboard_id=dashboard.id)
+            for dashboard in dashboards
+            if dashboard.id and not dashboard.hidden
+        }
+
+        # Get explore names from models
+        explores_for_model = {
+            model.name: [explore.name for explore in (model.explores or []) if explore.name]
+            for model in sdk.all_lookml_models()
+            if model.name
+        }
+
+        explores_by_id: Dict[str, LookmlModelExplore] = {}
+        for model_name, explore_names in explores_for_model.items():
+            for explore_name in explore_names:
+                try:
+                    lookml_explore = sdk.lookml_model_explore(
+                        lookml_model_name=model_name,
+                        explore_name=explore_name,
+                    )
+
+                    explores_by_id[check.not_none(lookml_explore.id)] = lookml_explore
+                except:
+                    logger.warning(
+                        f"Failed to fetch LookML explore '{explore_name}' for model '{model_name}'."
+                    )
+
+        return LookerInstanceData(
+            explores_by_id=explores_by_id,
+            dashboards_by_id=dashboards_by_id,
+        )
+
+    def to_cacheable_data(self, *, sdk: Looker40SDK) -> Sequence[AssetsDefinitionCacheableData]:
+        return [
+            AssetsDefinitionCacheableData(
+                extra_metadata={
+                    "dashboards_by_id": {
+                        dashboard_id: (
+                            sdk.serialize(api_model=dashboard_data).decode()  # type: ignore
+                        )
+                        for dashboard_id, dashboard_data in self.dashboards_by_id.items()
+                    },
+                    "explores_by_id": {
+                        explore_id: (
+                            sdk.serialize(api_model=explore_data).decode()  # type: ignore
+                        )
+                        for explore_id, explore_data in self.explores_by_id.items()
+                    },
+                }
+            )
+        ]
+
+    @staticmethod
+    def from_cacheable_data(
+        *, sdk: Looker40SDK, data: Sequence[AssetsDefinitionCacheableData]
```

Comment:
In a similar way here, could we avoid passing the `AssetsDefinitionsCacheableData`? We could pass the content of `entry.extra_metadata`